backend/src/routes/post.ts

```diff
@@ -25,7 +26,7 @@ router.get('/:postID', async (ctx, next) => {
 router.post('/', async (ctx, next) => {
 	const post = new Post();
 	post.id = new mongodb.ObjectId();
-	post.author = null; //This is dependent on Auth to determine the author
+	post.author = ctx.state.user.id;
```

Comment:
let's change this to store the username - you can grab it by finding the user model first via id then grabbing the username from the user model. you'll also need to change (in the post model file) the author field to be a string instead of a user model